client/my-sites/plans-features-main/hooks/use-plan-upgrade-credits-display.ts

```diff
@@ -0,0 +1,36 @@
+import { PlanSlug, PLAN_ENTERPRISE_GRID_WPCOM } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { usePlanUpgradeCredits } from 'calypso/my-sites/plans-features-main/hooks/use-plan-upgrade-credits';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { getSitePlanSlug } from 'calypso/state/sites/plans/selectors';
+import { isCurrentPlanPaid, isJetpackSite } from 'calypso/state/sites/selectors';
+
+export function usePlanUpgradeCreditsDisplay(
+	siteId: number,
+	visiblePlanNames: PlanSlug[] = []
+): {
+	creditsValue: number;
+	isPlanUpgradeCreditEligible: boolean;
+} {
+	const isSiteOnPaidPlan = !! useSelector( ( state ) => isCurrentPlanPaid( state, siteId ) );
+	const currentSitePlanSlug = useSelector( ( state ) => getSitePlanSlug( state, siteId ) );
+	const creditsValue = usePlanUpgradeCredits( siteId, visiblePlanNames );
+	const isJetpackNotAtomic = useSelector(
+		( state ) => isJetpackSite( state, siteId ) && ! isSiteAutomatedTransfer( state, siteId )
+	);
```

Comment:
NIT - This should really evolve out of components at some point, considering there are 7 or more definitions of the same in the codebase. Nothing to do here.